src/openrct2/ride/Vehicle.cpp

```diff
@@ -8286,38 +8283,39 @@ bool Vehicle::UpdateTrackMotionForwards(rct_ride_entry_vehicle* vehicleEntry, Ri
             remaining_distance = -1;
             return false;
         }
-        regs.ax = 0;
+        newTrackProgress = 0;
     }
 
-    track_progress = regs.ax;
+    track_progress = newTrackProgress;
     UpdateHandleWaterSplash();
 
     // loc_6DB706
     const auto moveInfo = GetMoveInfo();
     trackType = GetTrackType();
+    uint8_t moveInfovehicleSpriteType;
     {
         auto loc = TrackLocation
             + CoordsXYZ{ moveInfo->x, moveInfo->y, moveInfo->z + RideTypeDescriptors[curRide->type].Heights.VehicleZOffset };
 
-        regs.ebx = 0;
+        uint8_t flagsForSomething = 0;
```

Comment:
you've used different names for this later on. Be consistent. Its a flag for working out the track distance.